langchain/src/retrievers/multi_vector.ts

```diff
@@ -55,12 +60,44 @@ export class MultiVectorRetriever extends BaseRetriever {
   constructor(args: MultiVectorRetrieverInput) {
     super(args);
     this.vectorstore = args.vectorstore;
-    this.docstore = args.docstore;
+    if (args.byteStore) {
+      this.docstore = createDocumentStoreFromByteStore(args.byteStore);
+    } else if (args.docstore) {
+      this.docstore = args.docstore;
+    } else {
+      throw new Error(
+        "byteStore and docstore are undefined. Please provide at least one."
+      );
+    }
     this.idKey = args.idKey ?? "doc_id";
     this.childK = args.childK;
     this.parentK = args.parentK;
   }
 
+  /**
+   * Add documents to the byte store.
```

Comment:
This should also add index them in the vectorstore too

If it doesn't make sense to do so, we can leave it as `retriever.docstore.add()` above.